### PR TITLE
[node] almost fully synchronize node with node/ts4.8

### PR DESCRIPTION
**Pull request by jakebailey**
The only difference between the top level node types and the ts4.8 dir should be that one has DOM enabled, and one does not. In practice, it's not hard to accidentally forget to change both.

In #68371, I synchronized the `var global` declaration over into ts4.8. I noted that there were still things that were mistakenly not sync'd into the subdir but wanted to send the smaller change first. That smaller change seems to have worked.

This PR pulls over almost all of the remaining differences. For v20's types, the differences are:

```diff ts
diff --git a/./index.d.ts b/./ts4.8/index.d.ts
index a596cad021..7c8b38c634 100644
--- a/./index.d.ts
+++ b/./ts4.8/index.d.ts
@@ -22,7 +22,7 @@
  * IN THE SOFTWARE.
  */
 
-// NOTE: These definitions support NodeJS and TypeScript 4.9+.
+// NOTE: These definitions support NodeJS and TypeScript 4.8 and earlier.
 
 // Reference required types from the default lib:
 /// <reference lib="es2020" />
diff --git a/./node-tests.ts b/./ts4.8/node-tests.ts
index 1b5666a76a..7f7f1ed2f7 100644
--- a/./node-tests.ts
+++ b/./ts4.8/node-tests.ts
@@ -9,7 +9,7 @@ import "./test/crypto";
 import "./test/dgram";
 import "./test/diagnostics_channel";
 import "./test/dns";
-import "./test/dom-events"; // dom-events behaves differently under lib-dom
+import "./test/dom-events";
 import "./test/events";
 import "./test/fs";
 import "./test/globals";
diff --git a/./test/dom-events.ts b/./ts4.8/test/dom-events.ts
index d391c70851..635296c360 100644
--- a/./test/dom-events.ts
+++ b/./ts4.8/test/dom-events.ts
@@ -1,17 +1,16 @@
 //// Test global event interfaces
 
 {
-    const e: Event = new Event("");
-    e.preventDefault();
-    // @ts-expect-error - ensure constructor does not return a constructor
-    new e();
+    // Some event properties differ from DOM types
+    const evt = new Event("fake");
+    evt.cancelBubble();
+    // @ts-expect-error
+    evt.composedPath[2];
+    // $ExpectType 0 | 2
+    evt.eventPhase;
 
-    // Node's "DOM-like" Event differs from the real DOM in a few key aspects
-    // $ExpectType boolean
-    e.cancelBubble;
-    e.composedPath();
-    // $ExpectType number
-    e.eventPhase;
+    // @ts-expect-error - ensure constructor does not return a constructor
+    new evt();
 
     const et: EventTarget = new EventTarget();
     et.addEventListener("", () => {}, { passive: true });
diff --git a/./test/globals.ts b/./ts4.8/test/globals.ts
index bd361af5e2..333ceee110 100644
--- a/./test/globals.ts
+++ b/./ts4.8/test/globals.ts
@@ -54,19 +54,11 @@ declare var RANDOM_GLOBAL_VARIABLE: true;
 
 // fetch
 {
-    // This tsconfig.json references lib.dom.d.ts. The fetch
-    // types included in globals.d.ts are designed to be empty
-    // merges when lib.dom.d.ts is included. This test ensures
-    // the merge works, but the types observed are from lib.dom.d.ts.
     fetch("https://example.com").then(response => {
         response.arrayBuffer(); // $ExpectType Promise<ArrayBuffer>
         response.blob(); // $ExpectType Promise<Blob>
         response.formData(); // $ExpectType Promise<FormData>
-
-        // undici-types uses `Promise<unknown>` for `json()`
-        // This $ExpectType will change if tsconfig.json drops
-        // lib.dom.d.ts.
-        response.json(); // $ExpectType Promise<any>
+        response.json(); // $ExpectType Promise<unknown>
         response.text(); // $ExpectType Promise<string>
     });
     const fd = new FormData();
@@ -74,9 +66,7 @@ declare var RANDOM_GLOBAL_VARIABLE: true;
     const headers = new Headers();
     headers.append("Accept", "application/json");
     fetch("https://example.com", { body: fd });
-
     fetch(new URL("https://example.com"), {
-        // @ts-expect-error this should not be available when lib.dom.d.ts is present
         dispatcher: undefined,
     });
 
diff --git a/./test/perf_hooks.ts b/./ts4.8/test/perf_hooks.ts
index 669e1cdc72..f3a6183f5e 100644
--- a/./test/perf_hooks.ts
+++ b/./ts4.8/test/perf_hooks.ts
@@ -70,7 +70,6 @@ const mean: number = monitor.mean;
 const stddev: number = monitor.stddev;
 const exceeds: number = monitor.exceeds;
 
-// @ts-expect-error - Node API isn't available in DOM environment
 performance.eventLoopUtilization();
 
 let histogram: RecordableHistogram = createHistogram({
diff --git a/./tsconfig.json b/./ts4.8/tsconfig.json
index a181bc4bfb..7ce002fa66 100644
--- a/./tsconfig.json
+++ b/./ts4.8/tsconfig.json
@@ -7,8 +7,7 @@
         "module": "commonjs",
         "target": "esnext",
         "lib": [
-            "es6",
-            "dom"
+            "es6"
         ],
         "noImplicitAny": true,
         "noImplicitThis": true,
```

So, just those related to the DOM being different in testing.

For reference, I automated most of this process by running this command repeatedly until it stopped changing anything:

```
git diff --diff-filter=AM --no-index . ./ts4.8 | grep '^diff' | sed 's:a/::g' | sed 's:b/::g' | awk '{ print $3 " " $4 }' | head -n1 | xargs cp
```

Then per versioned folder. A little hacky, probably, but it worked.

### CODE REVIEW

File: types/node/v16/ts4.8/tsconfig.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
         "module": "commonjs",
         "target": "esnext",
         "lib": [
-            "dom"
+            "es6"
```
Comment (jakebailey) [inline types/node/v16/ts4.8/tsconfig.json:10]: This was totally wrong before.

File: types/node/v16/tsconfig.json [hunk 2]
```diff
@@ -7,6 +7,7 @@
         "module": "commonjs",
         "target": "esnext",
         "lib": [
+            "es6",
```
Comment (jakebailey) [inline types/node/v16/tsconfig.json:10]: As was this.

**Comment by jakebailey** [comment]
Ah, seems like a bug in dt-tools post-pnpm. FYI @sandersn

**Comment by jakebailey** [comment]
Seems like a conflict with https://github.com/microsoft/DefinitelyTyped-tools/pull/812; we need to not give this error when it's just a script changing.

**Comment by jakebailey** [comment]
Sent https://github.com/microsoft/DefinitelyTyped-tools/pull/929 which maybe fixes it.

**Comment by jakebailey** [comment]
I'm looking into the failures here to see if there's something up with dt-tools + pnpm.

**Comment by jakebailey** [comment]
My fix to the DT infra worked, so this PR is now passing.

**Review by johnnyreilly — APPROVED** [review]

**Comment by jakebailey** [comment]
Ready to merge

**Pull request merged**